docs/outgoing-webhooks.md

```diff
@@ -0,0 +1,114 @@
+# Outgoing Webhooks
+
+Outgoing webhooks are a type of integration where the Zulip server sends HTTP POST requests to a third party URL.
+To create an outgoing webhook, go to the `Add a new bot` tab in bot settings, and select the following options:
+* **Bot type**: outgoing webhook
+* **Full name, Username**: as desired, the username will be taken as name of outgoing webhook
+* **Base URL**: desired URL to which data is to be posted
+* **Interface**: generic for simple outgoing webhooks, else desired interface as per use
+
+
+![Add new bot form](images/add_new_bot_form.png)
+
+
+## Editing the details:
+**Base URL** and **Interface** for an outgoing webhook can be edited by users, through the `Edit bot` form.
+To open the `Edit bot` form, go to the `Active bots` tab, and for the outgoing webhook bot to be edited,
+click on the pencil shaped icon below the bot's name.
+
+
+![Edit bot form](images/edit_bot_form.png)
+
+
+## Triggering
+There are currently two methods to trigger an outgoing webhook:
+1.  **@-mention the bot which is the owner of outgoing webhook**
+
+    The bot you created through `Add new bot` ui is the owner of the outgoing webhook.
+    In this case the outgoing webhook will send the response in the same stream where it was triggered.
+    Note: make sure that the bot is subscribed to the stream where it is to be triggered.
+
+2.  **Send a private message to the bot**
+
+    This can be both 1:1 messages or a group message. The outgoing webhook would send the response in the
+    private message space where it was triggered.
+
+
+## Data posted
+The data posted varies according to the interface used. To know about the data posted to URL, refer to
+the section below.
+
+
+## Interfaces
+These are classes which customise the data to be posted to base URL of outgoing webhooks. These also provide
+methods to prepare data to be posted and help parse the response from the URL and generate the response
+to be sent to the user. For specialised purpose, user can create his own interface, else he can use
```

Comment:
This is a long sentence - perhaps use more commas? ("These also provide..." sentence)

In the next sentence, you could use "they" rather than "he" to stay neutral, as per the rest of the document ("user", "you").